builtin/providers/aws/resource_aws_spot_fleet_request.go

```diff
@@ -564,8 +583,28 @@ func resourceAwsSpotFleetRequestCreate(d *schema.ResourceData, meta interface{})
 		DryRun:                 aws.Bool(false),
 	}
 
-	log.Printf("[DEBUG] Requesting spot fleet with these opts: %+v", spotFleetOpts)
-	resp, err := conn.RequestSpotFleet(spotFleetOpts)
+	fmt.Printf("[DEBUG] Requesting spot fleet with these opts: %+v", spotFleetOpts)
+
+	// Since IAM is eventually consistent, we retry creation as a newly created role may not
+	// take effect immediately, resulting in an InvalidSpotFleetRequestConfig error
+	var resp *ec2.RequestSpotFleetOutput
```

Comment:
Wait a second, is this to make your itests pass?
I don't know about this... can you just add a "sleep" in your itest instead of putting this code in the resource itself?